### ISSUE TITLE
Not pulling all attributes

### ISSUE THREAD

**Issue by Jeffwww** [MscrmTools/MscrmTools.AttributeUsageInspector#13]
I used to tool to export attribute usage for 20 entities in D365, for mapping fields for a migration.  I found that while 17 entities were complete, 3 were not.  It was specifically the Quote, Order, and Invoice entities, which somewhat unique in D365, as entities go.  I found that I only got 184 attributes for the quote, despite seeing 241 in D365 customizations.  The numbers for the other two entities were similar.  I tried re-exporting, just the quote entity several times and it always comes back with 184 attributes.  I thought maybe it is something about those specific attributes, but found among the missing both simple text fields, and lookups, both custom and out of the box.  I can find no pattern to it, but it is the same fields that get excluded every time.  On the order I am missing over 100 fields.  I have exported other entities with far more fields and got exactly the right number.  Is there any explanation for this, or is there anything that I can do to get this to work correctly?
I am on the latest build of AttributeUsageInspector, pulling from CRM 2011 on prem.

**Comment by MscrmTools** [comment]
Seems weird but I will try to check the scenario myself. Well, not on an CRM 2011 on prem but still

**Comment by MscrmTools** [comment]
I guess there is a lot of custom attributes? I just have 91 attributes in quote entity in an online env

**Comment by Jeffwww** [comment]
Yes, there are quite a few custom fields, but not as many as some other entities that exported just fine.

________________________________

**Comment by Jeffwww** [comment]
FYI, using a different tool, Attribute Explorer, I see 133 out of the box fields on the quote.  Interestingly though, that tool gives me more than 300 total fields, even though I only see 241 in customizations.

**Comment by MscrmTools** [comment]
There are some virtual attributes that I don't take into account in my tool, that could explain differences

**Comment by Jeffwww** [comment]
The first one that I noticed was missing was just a normal money field.  It was a custom field, but nothing funny about it.  It is just a normal field that we populate by plugin, summing data up from child records.  No special properties of the field.

I'm going to try to identify what is missing the best I can and let you know if I can find some sort of pattern, or useful information, but it really just seems to be missing random fields.  Only on the Quote, Order, and Invoice entities though.

________________________________

**Comment by Jeffwww** [comment]
So there were quite a few virtual fields (ship to address fields, bill to address fields) but also pretty normal fields like source campaign (out of the box look up field).  I don't see anything unique about the fields that are missing, but they are consistently excluded.
